library/src/main/java/com/novoda/downloadmanager/lib/BatchRetrievalException.java

```diff
@@ -0,0 +1,15 @@
+package com.novoda.downloadmanager.lib;
+
+final class BatchRetrievalException extends RuntimeException {
+    static BatchRetrievalException failedQueryForBatches() {
+        return new BatchRetrievalException("Failed to query for batches");
+    }
+
+    static BatchRetrievalException failedQueryForBatch(long batchId) {
```

Comment:
like I showed the message should be inside the class and so people putting it outside is not a valid reason. With the above it enforces that you cannot have the full message as a param.

Creating an exception through a factory method is very strange, when something is exceptional you should know exactly why & what exception you are using.
If you want to separate these two exceptions by name i.e. they are that different - then they should be two different classes.